sky/task.py

```diff
@@ -621,18 +622,38 @@ def set_resources_override(self, override_params: Dict[str, Any]) -> 'Task':
     def service(self) -> Optional[service_spec.SkyServiceSpec]:
         return self._service
 
-    def set_service(self,
-                    service: Optional[service_spec.SkyServiceSpec]) -> 'Task':
+    def set_service(self, service: Optional[service_spec.SkyServiceSpec]):
         """Sets the service spec for this task.
-
         Args:
           service: a SkyServiceSpec object.
-
-        Returns:
-          self: The current task, with service set.
```

Comment:
Why removed? it is helpful for chain function call

Reply:
Do we require returning self somewhere?

Reply:
This is the tradition of setters for `sky.Task`.